File: src/main/java/me/miki/shindo/management/event/impl/EventRenderPlayer.java
package me.miki.shindo.management.event.impl;

import me.miki.shindo.management.event.Event;
import net.minecraft.entity.Entity;

public class EventRenderPlayer extends Event {

    private final Entity entity;
    private final double x;
    private final double y;
    private final double z;
    private final float partialTicks;

    public EventRenderPlayer(Entity entity, double x, double y, double z, float partialTicks) {
        this.entity = entity;
        this.x = x;
        this.y = y;
        this.z = z;
        this.partialTicks = partialTicks;
    }

    public Entity getEntity() {
        return entity;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public double getZ() {
        return z;
    }

    public float getPartialTicks() {
        return partialTicks;
    }
}

File: src/main/kotlin/com/shindoclient/shindo/management/event/impl/EventRenderPlayer.kt
package com.shindoclient.shindo.management.event.impl

import com.shindoclient.shindo.management.event.Event
import net.minecraft.entity.Entity

class EventRenderPlayer(
    private val entity: Entity,
    private val x: Double,
    private val y: Double,
    private val z: Double,
    private val partialTicks: Float,
) : Event() {
    fun getEntity(): Entity = entity

    fun getX(): Double = x

    fun getY(): Double = y

    fun getZ(): Double = z

    fun getPartialTicks(): Float = partialTicks
}
